rootfs/deploy.py

```diff
@@ -0,0 +1,48 @@
+import docker
+import os
+import tarfile
+import os.path
+import urllib
+import boto3
+
+from botocore.client import Config
+from urlparse import urlparse
+
+def log_output(stream):
+    for chunk in stream:
+        print(chunk)
+
+
+if os.getenv('TAR_URL'):
+    if os.path.exists("/var/run/secrets/object/store/access-key-id"):
+        with open("/var/run/secrets/object/store/access-key-id", "rb") as the_file:
+            secretId = the_file.read().replace('\n', '')
+        with open("/var/run/secrets/object/store/access-secret-key", "rb") as the_file:
+            secretKey = the_file.read().replace('\n', '')
+        url = urlparse(os.getenv('TAR_URL'))
+        hosturl = url.scheme + "://" + url.netloc
+        s3 = boto3.client('s3', config=Config(signature_version='s3v4'),
+                          endpoint_url=hosturl,
+                          aws_access_key_id=secretId,
+                          aws_secret_access_key=secretKey)
+        bucket = url.path.split('/')[1]
+        key = ('/').join(url.path.split('/')[2:])
+        s3.download_file(bucket, key, 'apptar')
+    else:
+        targzfile = urllib.URLopener()
+        targzfile.retrieve(os.getenv('TAR_URL'), "apptar")
+
+print "download tar file complete"
+tar = tarfile.open("apptar", "r:gz")
```

Comment:
Would prefer it if we used [with](https://www.python.org/dev/peps/pep-0343/) here so it ensures that the tar fd is closed